### DISCUSSION TITLE
multitrack podcast recording on teamtalk5

### DISCUSSION THREAD

**Discussion by MixingTech** [BearWare/TeamTalk5#1132] [Ideas]
Hi, hope all is well.
I'm an audio engineer recording podcasts through your TT record to hard drive system. I need more processing options in the mix so tried doing a multitrack recording with the separate audio file for each user function but then when it comes to mixdown the resulting files don't line up with each other, they just start when each user talks. can you please make it so that when start is pressed it records everything including silence and everything starts at the same time? For all users? This makes it a lot easier to line up and mix.
As stated, before I'm mixing podcasts so everyone will not be transmitting at the same time and some users will have voice activation enabled. 
My preferred mixing software is either Logic pro X or Pro Tools and I usually drop all the files in at the beginning of my software’s session so they should all start at 00:00:00 in the timeline.
I’m doing a lot of this work remotely so can’t just go and set up a mixing console, and I can’t think of any way to wire it in to my console with each user on his own fader if you understand me, as there's no way to root individual users to different outputs in the software.
Thanks in advance, 
Marshall

**Comment by Simon818** [comment]
This would be a problem for a few reasons. New users who join after recording starts would need a long file written for their user ID. There would be a LOT of extra data written to disc.

It would be helpful to have a timestamp that includes decimals. Then maybe a tool could be created to line the files up properly. I think Reaper and some other software has support for a log file that lists different audio tracks and where they should come in.

**Comment by MixingTech** [comment]
Hi. So you’re saying that once a user starts talking it would start recording their file and that file would have a timestamp of some description so I could line it up in Logic or Pro Tools which ever I was using at the time?So in a podcasting situation where a user contributes something for a minute then stops talking to allow other users to speak, would it then go on to start a new file when that user talks again or would it just keep recording to the same file.If it would just keep recording to the same file that would be OK since I could just line it up initially and I’d know that when my users start talking again the file will just continue so I can focus on volume / eq processing where neededIf a new user joins or has an internet disconnect, I usually have to start the recording again as it’ll throw off my gain staging so once someone new joins that’s the previous take effectively done Sent from Mail for Windows From: Simon JaegerSent: 13 November 2021 00:47To: BearWare/TeamTalk5Cc: MixingTech; AuthorSubject: Re: [BearWare/TeamTalk5] multitrack podcast recording on teamtalk5 (Discussion #1132) This would be a problem for a few reasons. New users who join after recording starts would need a long file written for their user ID. There would be a LOT of extra data written to disc.It would be helpful to have a timestamp that includes decimals. Then maybe a tool could be created to line the files up properly. I think Reaper and some other software has support for a log file that lists different audio tracks and where they should come in.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ChrisDuffley** [comment]
No.

Whenever voice activation/push-to-talk is disabled, the file is completely written and TeamTalk creates a new one when it's re-enabled.

The first part of the filename is the year/month/day, followed by the time and the ID/username of the client that was transmitting. For example, 20211001 142556 #8 chris.wav:
20211001 is October 1, 2021.
142556 is 14:25:56, or 2:25:56 PM.
#8 is the number of the client that was transmitting during that session since the TeamTalk server itself was started.
Chris is the username that was used for the person/device that was logged in and transmitting at that time. If no username is used for that recording, only a user ID will be shown.

Hope this helps!

Christopher Duffley

Host of Mission Possible Podcast <http://www.missionpossiblepodcast.com/>

LINKS TO SUBSCRIBE:

iTunes <https://itunes.apple.com/us/podcast/mission-possible-christopher/id1103092109?mt=2> | Stitcher <http://www.stitcher.com/s?fid=88094&refid=stpr> | Google Play <https://goo.gl/app/playmusic?ibi=com.google.PlayMusic&isi=691797987&ius=googleplaymusic&link=https://play.google.com/music/m/Ik76dexklcprwnpuxbdr2s6bm3y?t=Mission_Possible_with_Christopher_Duffley> | Follow Podcast On TuneIn And Listen To Latest Episode <http://tunein.com/radio/Mission-Possible-With-Christopher-Duffley-p892505/>

LIKE/FOLLOW Me:

Facebook <http://facebook.com/christopherduffley> | Twitter <http://twitter.com/chrisduffley> | YouTube <http://youtube.com/christopherduffley> | Instagram <http://instagram.com/christopherduffley>

On Nov 13, 2021, at 05:16, MixingTech ***@***.***> wrote:


Hi. So you’re saying that once a user starts talking it would start recording their file and that file would have a timestamp of some description so I could line it up in Logic or Pro Tools which ever I was using at the time?So in a podcasting situation where a user contributes something for a minute then stops talking to allow other users to speak, would it then go on to start a new file when that user talks again or would it just keep recording to the same file.If it would just keep recording to the same file that would be OK since I could just line it up initially and I’d know that when my users start talking again the file will just continue so I can focus on volume / eq processing where neededIf a new user joins or has an internet disconnect, I usually have to start the recording again as it’ll throw off my gain staging so once someone new joins that’s the previous take effectively done Sent from Mail for Windows From: Simon JaegerSent: 13 November 2021 00:47To: BearWare/TeamTalk5Cc: MixingTech; AuthorSubject: Re: [BearWare/TeamTalk5] multitrack podcast recording on teamtalk5 (Discussion #1132) This would be a problem for a few reasons. New users who join after recording starts would need a long file written for their user ID. There would be a LOT of extra data written to disc.It would be helpful to have a timestamp that includes decimals. Then maybe a tool could be created to line the files up properly. I think Reaper and some other software has support for a log file that lists different audio tracks and where they should come in.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by bear101** [comment]
@MixingTech I would suggest you use the TeamTalk SDK to develop this feature. With [audio-block callback](http://www.bearware.dk/teamtalksdk/v5.8b/docs/C-API/group__sounddevices.html#ga332b045b503ea31646fd26072e0e6da2) you can access the raw audio of each user and mix it as you like.

**Comment by Simon818** [comment]
I agree with your solution, but could you actually add milliseconds to recorded file names? This would allow for some precision if someone did want to manually align the files. Right now there's a two-second error margin.

**Comment by bear101** [comment]
Can't you just use the file-create timestamp?

**Comment by Simon818** [comment]
I didn't actually think that had millisecond precision (at least on Windows). It would also be subject to any delays on file i/o. TeamTalk knows the precise moment when someone starts transmitting.

**Reply by bear101** [reply]
The file timestamp has 100ns precision, so it's every more precise than when TT submits the audio buffer to the sound device.

**Comment by MixingTech** [comment]
Hi, sorry for being inactive haven’t forgotten about this issue. Replying to the suggestion about using the TT SDK to develop an audio feature to mix the raw tracks:Unfortunately I don’t have my hand in coding. I’m used to working in a physical situation with mics, a board, or interface, and a pair of headphones, will mix in logic or PT.The way things are at the minute I use a redundancy system which is usually running the whole show and I have to record to 1 track only, meaning if you don’t set your levels correct you’re kind of fucked and it can prove difficult to fix afterwards.I’d be a fan of recording with multitracks if the voice activation causing it to create a new file were not an issue.If you used the TT SDK what would that do? Would it be a piece of code that basically enables me to have the raw unedited tracks for each user available to me so I can drop them in to the session?ThanksMarshall Sent from Mail for Windows From: Bjørn Damstedt RasmussenSent: 15 November 2021 19:50To: BearWare/TeamTalk5Cc: MixingTech; MentionSubject: Re: [BearWare/TeamTalk5] multitrack podcast recording on teamtalk5 (Discussion #1132) The file timestamp has 100ns precision, so it's every more precise than when TT submits the audio buffer to the sound device.—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
